Qa add test pcp 4366 4367 4368 4369 4370 4371 4372 4373

sedatbsr opened the pull request:
Tests added:
PCP-4366 | Settings - US - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - UK - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - Canada - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - France - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - Italy - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - Spain - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - Australia - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen
PCP-4366 | Settings - Germany - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen

Commit:
PCP-4366-4367-4368-4369-4370-4371-4372-4373 added

Commit:
Merge branch 'qa' into QA-add-test-pcp-4366-4367-4368-4369-4370-4371-4372-4373

Commit:
fix: PCP-4366,4367,4368,4369,4370,4371,7372,4373

function for screenshot and a locator removed from pcp-onboarding

mishautkin committed:
Update .gitignore

mishautkin committed:
Add locators for Onboarding page

mishautkin committed:
Add/update snapshot methods in `pcp-admin-page`

mishautkin committed:
Remove `percy` from resources

mishautkin committed:
Replace `percy` in subscriptions

mishautkin committed:
Update `visual-onboarding-ui.spec`

- Replace percy
- Move out badges tests

mishautkin committed:
Add separate spec with badges tests

mishautkin committed:
General changes

mishautkin committed:
Update Playwright Screenshots functionality

- Update options argument and type in pcp-admin-page.snapshotLocator
- Update snapshot methods in PLM, Payment Methods and Styling POMs

Commit:
Updates

Extra test block for Germany removed
Test script updated to be dynamic to cover the Germany case
Test data file updated.

Commit:
.only removed

Commit:
Merge branch 'qa' into QA-add-test-pcp-4366-4367-4368-4369-4370-4371-4372-4373

mishautkin commented on tests/qa/tests/02-dashboard-ui/visual-onboarding-ui.spec.ts:
```diff
@@ -133,26 +133,15 @@ test.describe.only('', () => {
-		await pcpOnboarding.businessRadio().click();
-		await pcpOnboarding.continueButton().click();
+		const businessRadio = await pcpOnboarding.businessRadio();
+		if (await businessRadio.isVisible()) {
```
Please consider that `.isVisible()` has immediate action and doesn't wait for timeout. If the element is not visible at the time you call `.isVisible()` - the condition will result in `false`.

IMO for this reason better to use country-based condition `if( country !== 'Germany' )`.

sedatbsr replied:
![QA 2025-05-09 at 05 39 - Google Chrome](https://github.com/user-attachments/assets/81f5c495-acf7-446c-9c1a-50162179a152)

Right. I overlooked this. I will update the test and the data accordingly.

mishautkin commented on tests/qa/tests/02-dashboard-ui/visual-onboarding-ui.spec.ts:
```diff
@@ -117,7 +117,7 @@ test( 'PCP-4318 | Settings - US - Onboarding - Connect with business account, al
 } );
 
-test.describe.only('', () => {
+test.describe.only('Comparison of initial and checkout pages of onboarding per country', () => {
```
Point of view.
This long string duplicates info from test summary and will appear in console, limiting space of test summary itself or forcing it to new line.

I know that we're using both formats of test data:
- with only `testKey`
- with full `testSummary`

The latter reduces readability of test.spec making it hard to see what test case is about. Maybe we should stick to only providing `testKey` in the test data? This way we also avoid hardcoding strings. What is your opinion?
```ts
`${ testKey } | Settings - ${ testCountry } - Onboarding - Compare initial onboarding page (right part) with expanded checkout screen`

sedatbsr replied:
I didn't think about the part you mentioned. But I thought since we started to use PW SS and the testKey itself will not be enough to understand what that screenshot is for. 
Anyway, in terms better visibility, this can be done.

Commit:
Fix and changes

Country and testKey added to test data
Test summary removed
Test script updated accordingly
Fix lint issues

Commit:
Fix and changes

Country and testKey added to test data
Test summary removed
Test script updated accordingly
Fix lint issues
